docs/attestation-storage.md

```diff
@@ -0,0 +1,212 @@
+  manifest for a selected image manifest.
+  
+### Attestation Manifest
+
+All properties of standard OCI and Docker manifest descriptor continue to
+apply.
+
+The image `config` property will be set as normal, and contains a valid image
+config descriptor. The image config content will be as minimally valid as
+possible, containing only platform properties (set to `unknown/unknown` as
+above), and the `diff_ids` for the layers.
+
+```json
+{
+  "architecture": "unknown",
+  "os": "unknown",
+  "config": {},
+  "rootfs": {
+    "type": "layers",
+    "diff_ids": [
+      ...
+    ]
+  }
+}
+```
+
+The image `layers` will each contain a single attestation. The `mediaType` of
+each layer will be set in accordance to its contents, one of:
+
+- `application/vnd.in-toto+json` (currently, the only supported option)
+
+  Indicates an [in-toto attestation layer](#attestation-in-toto-layer).
+
+Any unknown `mediaType`s should be ignored.
+
+To assist attestation traversal, the following annotations may be set on each
+layer descriptor:
+
+- `in-toto.io/predicate-type`
+
+  This annotation will be set if the enclosed attestation is an in-toto
+  attestation (currently, the only supported option). The annotation will
+  be set to contain the same value as the `predicateType` property present
+  inside the attestation.
+  
+### Attestation in-toto Layer
+
+The layer contents will contain a full [in-toto attestation statement](https://github.com/in-toto/attestation/blob/main/spec/README.md#statement):
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "subject": [
+    {
+      "name": "<NAME>",
+      "digest": {"<ALGORITHM>": "<HEX_VALUE>"}
+    },
+    ...
+  ],
+  "predicateType": "<URI>",
+  "predicate": { ... }
+}
+```
+
+The subject of the attestation should be set to be the same digest as the
+target manifest described in the [Attestation Manifest Descriptor](#attestation-manifest-descriptor), or some object within.
+
+## Examples
+
+*Example showing an SBOM attestation attached to a `linux/amd64` image*
+
+Image index:
+
+```json
+{
+  "mediaType": "application/vnd.oci.image.index.v1+json",
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:23678f31b3b3586c4fb318aecfe64a96a1f0916ba8faf9b2be2abee63fa9e827",
+      "size": 1234,
+      "platform": {
+        "architecture": "amd64",
+        "os": "linux"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:02cb9aa7600e73fcf41ee9f0f19cc03122b2d8be43d41ce4b21335118f5dd943",
+      "size": 1234,
+      "annotations": {
+        "vnd.docker.reference.digest": "sha256:23678f31b3b3586c4fb318aecfe64a96a1f0916ba8faf9b2be2abee63fa9e827",
+        "vnd.docker.reference.type": "attestation-manifest"
+      },
+      "platform": {
+         "architecture": "unknown",
+         "os": "unknown"
+      }
+    }
+  ]
+}
+```
+
+Attestation manifest:
```

Comment:
Put digests for each of these objects so it is easier to follow what object it is.